### PR TITLE
 raise on try

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

same raise_on_try feature as we introduced in open-aea. Cannot be merged before next open-aea release

https://github.com/valory-xyz/open-aea/pull/197
https://github.com/valory-xyz/open-aea/pull/209

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by DavidMinarsch — DISMISSED** [review]
LGTM - let's wait for the next `open-aea` release before merging.

**Comment by Karrenbelt** [comment]
CI failures:

```
FAILED tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum[gas_strategies1]
FAILED tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum[gas_strategies2]
FAILED tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_get_balance[ethereum-0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266]
FAILED tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_get_state[ethereum-0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266]
FAILED tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum[gas_strategies0]
FAILED tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum[gas_strategies3]
FAILED tests/test_skills/test_liquidity_provision_abci/test_behaviours_integration.py::TestLiquidityRebalancingHardhat::test_full_run
FAILED tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py::TestSyncing::test_sync_local_hash
FAILED tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py::TestRepricing::test_same_keeper[True]
FAILED tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py::TestRepricing::test_same_keeper[False]
```

locally failing
```
tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum[gas_strategies0]
tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py::TestRepricing::test_same_keeper[True]
tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py::TestRepricing::test_same_keeper[True]
```
other failures might be side effect of others

File: tests/test_connections/test_ledger/test_ledger_api.py
Comment (Karrenbelt) [inline tests/test_connections/test_ledger/test_ledger_api.py:366]: this resolved the non-deterministic failure in `tests/test_connections/test_ledger/test_ledger_api.py::TestLedgerConnection::test_send_signed_transaction_ethereum`

Comment (Adamantios) [reply tests/test_connections/test_ledger/test_ledger_api.py:366]: Could you explain why this solves the issue? Why do we need to wait here?

Comment (Karrenbelt) [reply tests/test_connections/test_ledger/test_ledger_api.py:366]: I thought about it, and I'm not quite sure why this is now needed whereas it was not before. I can explain my reasoning to some extend: the failures were non-deterministic and, since the async is the only source of randomness, some "instinct" perhaps, for lack of a better word, led me to add a bit of sleep here. I realize this answer is not satisfactory but it's all I have :(

File: tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py
Comment (DavidMinarsch) [reply tests/test_skills/test_transaction_settlement_abci/test_failure_modes.py:270]: Then it needs to be fixed. This is not mergeable until the existing features work with it!

File: packages/valory/connections/ledger/ledger_dispatcher.py
Comment (Adamantios) [inline packages/valory/connections/ledger/ledger_dispatcher.py:273]: This is the reason that the repricing test fails. Previously, it would just sleep and loop until settling, if attempts were not exceeded. Now, it will return an error message and loop. Therefore, we need to either:
1. Update the test in order to sleep before `get_transaction_receipt` to avoid getting error messages. 
2. Account for a number of error messages.

Getting an error message before the transaction settles is normal.

File: tests/test_connections/test_ledger/test_ledger_api.py
Comment (Adamantios) [reply tests/test_connections/test_ledger/test_ledger_api.py:366]: I think this is the reason:

https://github.com/valory-xyz/open-autonomy/pull/1025#discussion_r916499210

File: packages/valory/connections/ledger/ledger_dispatcher.py
Comment (Adamantios) [reply packages/valory/connections/ledger/ledger_dispatcher.py:273]: Fixed in fcf5341fddd15b2ea92d49c44448b7a05352801a.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**
